src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -188,25 +172,14 @@ export class Kernel implements IKernel {
             cellHashProviderFactory
         );
         this.kernelExecution.onPreExecute((c) => this._onPreExecute.fire(c), this, disposables);
-        const isPreferredKernel =
-            getResourceType(resourceUri) === 'notebook'
-                ? notebookControllerManager.getPreferredNotebookController(this.notebookDocument)?.controller ===
-                  controller
-                : undefined;
-        if (pythonChecker.isPythonExtensionInstalled) {
-            trackKernelResourceInformation(resourceUri, {
-                kernelConnection: kernelConnectionMetadata,
-                isPreferredKernel
-            });
-        }
     }
     private perceivedJupyterStartupTelemetryCaptured?: boolean;
 
-    public addEventHook(hook: (event: 'willRestart' | 'willInterrupt') => Promise<void>): void {
+    public addEventHook(hook: (kernel: IKernel, event: 'willRestart' | 'willInterrupt') => Promise<void>): void {
```

Comment:
Sending the `IKernel` feels redundant, when the calling code can easily capture that object in context and use it.

Reply:
I like it this way (follows the .NET event way of doing things) but it doesn't really matter. I can remove it too.

Reply:
Thanks, sorry, but would prefer to change to standard node style, as this is very much non standard, e.g. even in VS Codes style for events (event emitter) we don't send the `sender`.